control-plane/agents/src/bin/core/node/wrapper.rs

```diff
@@ -841,7 +857,16 @@ impl NodeStateFetcher {
         let pools = self.fetch_pools(client).await?;
         let nexuses = self.fetch_nexuses(client).await?;
         let snapshots = self.fetch_snapshots(client).await?;
-        Ok((replicas, pools, nexuses, snapshots))
+        let rebuild_history = self.fetch_rebuild_history(client).await?;
```

Comment:
just realized there may be an issue here, also exists with snapshots. if the dataplane does not support this call, we'll get grpc error..
We need to ignore this error, but where, here, or in the node wrapper? @Abhinandan-Purkait wdyt?

Reply:
But we already return unimplemented for v0 right? So do we need to do anything more? 

Are you talking about the older versions on v1? For that I think we should do that in the node wrapper, it would be easy to manage from one place?

Reply:
For both cases... otherwise we'll fail this procedure here with `?`.
I'm not sure, the disadvantage of doing in the node wrapper is that we'd be hiding the errors from the caller.
Tempted to say for now let's handle it here, ignore not implemented error inline.

Reply:
Right, that's true. Then in that case this place is better right. The other callers will still see the errors which is fine as it's unimplemented. And here we handle it.

@tiagolobocastro A question, what happens if any of the above calls fail for whatever reason, then the below ones would not get executed right? Should we not attempt all calls and return an aggregated error at the end?

Reply:
hmm don't think that would help, we mostly just care if it succeeded or not. If any fails we just bail out and set the node as bad.

Reply:
But is this behaviour correct then? What if it was some interim failure and there is a chance that the other calls would have passed? Also should we not set the node as bad only on `transport error not route to host` something like that rather than for any error.

Reply:
Yes, because we don't want to keep half states, let's say you were able to list replicas but not pools, control-plane could get very confused.

Reply:
Got it. Thanks for explaining.